### ISSUE TITLE
salt-ssh --thin does not refresh python modules

### ISSUE THREAD

**Issue by max-arnold** [saltstack/salt#48486]
### Description of Issue/Question

I noticed that my modifications of builtin saltstack top modules do not propagate even if I use `salt-ssh --thin example.com state.highstate`

The modules are synced only when I remove all local `*.pyc` files: `find ~/.virtualenvs/salt/lib/python3.6/site-packages/salt/ -name '*.pyc' -delete`

I do not see any `*.pyc` files on the remote host in `/var/tmp/.root_7e26e1_salt/py3/salt/tops` folder.

However, my local thin archive has duplicate `*.py` files, and this may be the cause:

```
tar -tf var/cache/salt/master/thin/thin.tgz | sort | grep salt/tops/
py3/salt/tops/__init__.py
py3/salt/tops/__init__.py
py3/salt/tops/cobbler.py
py3/salt/tops/cobbler.py
py3/salt/tops/ext_nodes.py
py3/salt/tops/ext_nodes.py
py3/salt/tops/mongo.py
py3/salt/tops/mongo.py
py3/salt/tops/reclass_adapter.py
py3/salt/tops/reclass_adapter.py
py3/salt/tops/saltclass.py
py3/salt/tops/saltclass.py
py3/salt/tops/varstack.py
py3/salt/tops/varstack.py
```

### Setup

Salt-ssh is running on MacOS, remote host is Ubuntu 18.04.

### Versions Report

<details><summary>salt --versions-report</summary>
Salt Version:
           Salt: 2018.3.2

Dependency Versions:
           cffi: 1.11.5
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.6.5 (default, Mar 29 2018, 15:37:32)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 17.0.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.1.6

System Versions:
           dist:
         locale: UTF-8
        machine: x86_64
        release: 17.6.0
         system: Darwin
        version: 10.13.5 x86_64
</details>

**Comment by max-arnold** [comment]
Any news on this? I'm still observing duplicate modules in the thin archive made with `2018.3.3`.

**Comment by max-arnold** [comment]
Bump

**Comment by max-arnold** [comment]
Bump

**Comment by max-arnold** [comment]
Bump

**Comment by sagetherage** [comment]
@max-arnold this stale bot shouldn't be an issue anymore

**Comment by sagetherage** [comment]
I am going to kick this back to triage to review it, again and I will follow up with triage on Monday, 2020-04-13

**Comment by max-arnold** [comment]
Some archaeology regarding these duplicate `*.py` files:

```
tar -tf var/cache/salt/master/thin/thin.tgz | sort | grep salt/tops/
py3/salt/tops/__init__.py
py3/salt/tops/__init__.py
py3/salt/tops/cobbler.py
py3/salt/tops/cobbler.py
py3/salt/tops/ext_nodes.py
py3/salt/tops/ext_nodes.py
py3/salt/tops/mongo.py
py3/salt/tops/mongo.py
py3/salt/tops/reclass_adapter.py
py3/salt/tops/reclass_adapter.py
py3/salt/tops/saltclass.py
py3/salt/tops/saltclass.py
py3/salt/tops/varstack.py
py3/salt/tops/varstack.py
```

They are added in `salt/utils/thin.py:get_tops()`:

```
print(tops)
['/Users/user/.virtualenvs/salt2/lib/python3.7/site-packages/salt',  # <-- 1st occurrence
'/Users/user/.virtualenvs/salt2/lib/python3.7/site-packages/jinja2',
'/Users/user/.virtualenvs/salt2/lib/python3.7/site-packages/yaml',
'/Users/user/.virtualenvs/salt2/lib/python3.7/site-packages/tornado',
'/Users/user/.virtualenvs/salt2/lib/python3.7/site-packages/msgpack',
'/Users/user/.virtualenvs/salt2/lib/python3.7/site-packages/salt/ext/six.py',
'/Users/user/.virtualenvs/salt2/lib/python3.7/site-packages/salt/ext/backports_abc.py',
'/Users/user/.virtualenvs/salt2/lib/python3.7/site-packages/certifi',
'/Users/user/.virtualenvs/salt2/lib/python3.7/site-packages/salt',   # <-- 2nd occurrence (added by "if HAS_SSL_MATCH_HOSTNAME" statement)
'/Users/user/.virtualenvs/salt2/lib/python3.7/site-packages/markupsafe']
```

`ssl_match_hostname.__file__` equals to `/Users/user/.virtualenvs/salt2/lib/python3.7/site-packages/salt/ext/ssl_match_hostname.py` and after two `os.path.dirname` calls results in `/Users/user/.virtualenvs/salt2/lib/python3.7/site-packages/salt`. That is why every file under `site-packages/salt` gets added to `thin.tgz` twice.

The `if HAS_SSL_MATCH_HOSTNAME` statement was introduced in https://github.com/saltstack/salt/pull/19321

It only affects 2018.3.x and doesn't happen in `2019.2.x` and `3000.x`, because that part of code was heavily refactored in https://github.com/saltstack/salt/pull/46684

Now about the `thin.tgz` refresh issue. It happened on 2018.3.2 and seems to be fixed in 2018.3.3 (see https://github.com/saltstack/salt/pull/48636).

Since the 2018.3.x is going EOL on Oct 31, 2020, I guess it makes sense to just close the issue.
